contracts/base/ModuleManager.sol

```diff
@@ -463,6 +458,12 @@ abstract contract ModuleManager is Storage, EIP712, IModuleManagerEventsAndError
         return _getAccountStorage().validators.contains(validator);
     }
 
+    /// @dev Checks if there is at least one validator installed.
+    /// @return True if there is at least one validator, otherwise false.
+    function _hasValidators() internal view returns (bool) {
+        return _getAccountStorage().validators.getNext(address(0x01)) != address(0x01);
```

Comment:
I just checked the `SentinelListLib` again. There's a bug in this lib when you call `popAll()` it leaves the list in an uninitialized state instead of an initialized but empty state.

You're not using `popAll()` but it's better to fix it and change this check to also ensure the `next` is not the `zero` address (uninitialized):

```solidity
return _getAccountStorage().validators.getNext(address(0x01)) != address(0x01) && _getAccountStorage().validators.getNext(address(0x01)) != address(0x00);
```

Reply:
would this comment be appropriate then? @MrToph 

// Sentinel pointing to itself means the list is empty, so check this after removal

Reply:
will merge this PR and tackle in https://github.com/bcnmy/nexus/pull/121

Reply:
just double check above comment. actual fix is done @MrToph

Reply:
You can write this instead

> // Sentinel pointing to itself / zero means the list is empty / uninitialized, so check this after removal

Reply:
ok changing this